components/Pricing.js

```diff
@@ -1,37 +1,41 @@
 import cn from 'classnames';
 import { useRouter } from 'next/router';
 import { useState } from 'react';
-import { postData } from '../utils/helpers';
-import { getStripe } from '../utils/initStripejs';
-import { useUser } from '../components/UserContext';
-import Button from './ui/Button';
+
+import Button from '@/components/ui/Button';
+import { postData } from '@/utils/helpers';
+import { getStripe } from '@/utils/stripe-client';
+import { useUser } from '@/utils/useUser';
 
 export default function Pricing({ products }) {
-  const [billingInterval, setBillingInterval] = useState('month');
-  const [loading, setLoading] = useState(false);
   const router = useRouter();
+  const [billingInterval, setBillingInterval] = useState('month');
+  const [priceIdLoading, setPriceIdLoading] = useState();
   const { session, userLoaded, subscription } = useUser();
 
   const handleCheckout = async (price) => {
-    setLoading(true);
+    setPriceIdLoading(price.id);
     if (!session) {
-      router.push('/signin');
-      return;
+      return router.push('/signin');
     }
     if (subscription) {
-      router.push('/account');
-      return;
+      return router.push('/account');
+    }
+
+    try {
+      const { sessionId } = await postData({
+        url: '/api/create-checkout-session',
+        data: { price },
+        token: session.access_token
+      });
+
+      const stripe = await getStripe();
+      stripe.redirectToCheckout({ sessionId });
+    } catch (error) {
+      return alert(error.message);
+    } finally {
+      setPriceIdLoading(false);
     }
-    const { sessionId, error: apiError } = await postData({
-      url: '/api/createCheckoutSession',
-      data: { price },
-      token: session.access_token
-    });
-    if (apiError) return alert(apiError.message);
-    const stripe = await getStripe();
-    const { error: stripeError } = stripe.redirectToCheckout({ sessionId });
```

Comment:
Will this throw if there's an error? I was trying to define a better pattern here where all errors are handled in the catch block.